### ISSUE TITLE
[Question] Interaction between batch size and input chunk length

### ISSUE THREAD

**Issue by lams0018** [unit8co/darts#1699]
Hi, I'm wondering how batches of data are selected when we indicate the batch size for training the darts models, as well as how it's related to the input chunck length.

Qn 1)
For example in a batch size of 32 out of 1,000 training samples, does darts take a sequential rolling window approach i.e. 
first batch = samples 1 to 32
second batch = samples 2 to 33
etc

or 

would the training samples be taken in blocks of 32 i.e.
first batch = samples 1 to 32
second batch = samples 33 to 64
etc 

Furthermore if it's taken in blocks of 32, would the selection be randomized among different blocks of data?
Have tried to look under the documentation in PyTorch Lightning Trainers but it wasn't so clear to me. Appreciate if you could help to clarify.

Qn2) 
If for example our batch size is 32, while input chunk length is 14, does it mean that only a maximum of 28 samples are used for training in a batch?

Thank you

**Comment by madtoinou** [comment]
Hi @lams0018,

Q1) To train any Torch based (Deep Learning) mode, a Torch `DataLoader` is created around the Darts `Dataset`: the `batch_size` argument correspond to the number of samples for each forward pass in the gradient descent (traditional meaning) whereas `input_chunk_length` is specific to darts; it's the number of Timestamps in each sample (i.e. the number of tokens in a sequence or number of letters in a word).

By default, the training samples are shuffled by the Torch `DataLoader`, and the sampling strategy matches your second description. This shuffling occurs at each epoch.

You can find the relevant code snippet [here](https://github.com/unit8co/darts/blob/master/darts/models/forecasting/torch_forecasting_model.py#L978-L986)

Q2) If you pass `batch_size=32` and `input_chunk_length=14`, every batch will contain 32 samples containing 14 values ie the shape should be be [32, 14].  If your total number of training sample is not divisible by the batch size, the last batch will contain the remaining samples (because `drop_last` is set to `True` in the `DataLoader`).

**Comment by lams0018** [comment]
Thanks for the clarification!
